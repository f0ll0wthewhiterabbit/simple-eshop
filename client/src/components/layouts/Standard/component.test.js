import React from 'react'
import { shallow } from 'enzyme'

import StandardLayout from './component'
import findByTestAttr from '../../../utils/findByTestAttr'

describe('StandardLayout component', () => {
  let props
  let wrapper
  const testContent = 'test content'
  const testChildren = <p>{testContent}</p>

  const generateWrapper = passedProps => {
    const initialProps = {
      requestUserDeletion: jest.fn(),
      children: testChildren,
    }
    props = { ...initialProps, ...passedProps }

    return shallow(<StandardLayout {...props} />)
  }

  beforeEach(() => {
    wrapper = generateWrapper()
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should contain correct children in main', () => {
    const main = findByTestAttr(wrapper, 'main')

    expect(main.find('p').text()).toEqual(testContent)
  })
})
