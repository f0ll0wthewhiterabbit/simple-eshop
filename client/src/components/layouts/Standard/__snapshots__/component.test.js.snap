// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StandardLayout component should render correctly 1`] = `
<styled.div>
  <Connect(AlertDialog)
    confirmMethod={[MockFunction]}
    title="Delete your account?"
  >
    The remove request will be sent to the administrator. Unfortunately, your account will be deleted soon.
  </Connect(AlertDialog)>
  <Connect(Header) />
  <styled.main
    data-test="main"
  >
    <p>
      test content
    </p>
  </styled.main>
  <Footer />
</styled.div>
`;
